src/lib/cfa/nm_connection.rb

```diff
@@ -31,15 +35,26 @@ class NmConnection < BaseModel
     KNOWN_SECTIONS = [
       "bridge", "connection", "ethernet", "ipv4", "ipv6", "vlan", "wifi", "wifi_security"
     ].freeze
+    SYSTEM_CONNECTIONS_DIR = "/etc/NetworkManager/system-connections".freeze
 
+    class << self
+      # Returns all connection definition files
+      #
+      # @return [Array<NmConnection>]
+      def all
+        directory = File.join(Yast::Installation.destdir, SYSTEM_CONNECTIONS_DIR)
+        files = Dir[File.join(directory, "*.nmconnection")].to_a
+        files.map { |f| new(f) }
+      end
+    end
     # Constructor
     #
     # @param path [String] File path
     # @param file_handler [.read, .write] Object to read/write the file.
     def initialize(path, file_handler: nil)
       # FIXME: The Networkmanager lense writes the values surrounded by double
       # quotes which is not valid
-      super(AugeasParser.new("Puppet.lns"), path, file_handler: file_handler)
+      super(AugeasParser.new("Desktop.lns"), path, file_handler: file_handler)
```

Comment:
The puppet lense was ok for writting the config but not for reading, @mchf was taking care of creating an issue in the augeas repository.

Reply:
@imobachgs @mchf we were already evaluating the use of desktop instead of the puppet.lns when adding read support. Probably we should do the same change in #1194